featuretools/selection/selection.py

```diff
@@ -18,3 +18,115 @@ def remove_low_information_features(feature_matrix, features=None):
+                If no feature list is provided as input, the feature list will not be returned.
+    """
+    unique_counts_by_col = feature_matrix.nunique(dropna=not count_nan_as_value).to_dict()
+
+    keep = [f_name for f_name, unique_count in unique_counts_by_col.items() if unique_count > 1]
+    return _apply_feature_selection(keep, feature_matrix, features)
+
+
+def remove_highly_correlated_features(feature_matrix, features=None, pct_corr_threshold=0.95, features_to_check=None, features_to_keep=None):
+    """Removes columns in feature matrix that are highly correlated with another column.
+
+        Args:
+            feature_matrix (:class:`pd.DataFrame`): DataFrame whose columns are feature names and rows are instances.
+            features (list[:class:`featuretools.FeatureBase`] or list[str], optional): List of features to select.
+            pct_corr_threshold (float): The correlation threshold to be considered highly correlated. Defaults to 0.95.
+            features_to_check (list[str], optional): List of column names to check whether any pairs are highly correlated.
+                        Will not check any other columns.
+                        If null, defaults to checking all columns.
+            features_to_keep (list[str], optional): List of colum names to keep even if correlated to another column.
+                        If null, all columns will be candidates for removal.
+
+        Returns:
+            pd.DataFrame, list[:class:`.FeatureBase`]:
+                The feature matrix and the list of generated feature definitions. Matches dfs output.
+                If no feature list is provided as input, the feature list will not be returned.
+    """
+    if pct_corr_threshold < 0 or pct_corr_threshold > 1:
+        raise ValueError("pct_corr_threshold must be a float between 0 and 1, inclusive.")
+
+    if features_to_check is None:
+        features_to_check = feature_matrix.columns
+    else:
+        for f_name in features_to_check:
+            assert f_name in feature_matrix.columns, "feature named {} is not in feature matrix".format(f_name)
+
+    if features_to_keep is None:
+        features_to_keep = []
+
+    numeric_dtypes = ['int16', 'int32', 'int64', 'float16', 'float32', 'float64']
+    boolean = ['bool']
+    numeric_and_boolean_dtypes = numeric_dtypes + boolean
+
+    fm_to_check = (feature_matrix[features_to_check]).select_dtypes(include=numeric_and_boolean_dtypes)
+
+    # Get all pairs of columns and calculate their correlation, dropping any columns that are highly correlated
+    dropped = set()
+    for f_name1, col1 in fm_to_check.iteritems():
+        for f_name2, col2 in fm_to_check.iteritems():
+            if f_name1 == f_name2 or f_name1 in dropped or f_name2 in dropped:
+                continue
+
+            if abs(col1.corr(col2)) >= pct_corr_threshold:
+                dropped.add(f_name1)
```

Comment:
We don't want to drop both features -- we want to keep one and drop the other.

Reply:
Changed to keep the less complex of the two features, which we'll treat as the feature that comes earlier in the list